lib/database_cleaner/safeguard.rb

```diff
@@ -0,0 +1,70 @@
+module DatabaseCleaner
+  class Safeguard
+    class Error < Exception
+      class RemoteDatabaseUrl < Error
+        def initialize
+          super("ENV['DATABASE_URL'] is set to a remote URL. Please refer to https://github.com/DatabaseCleaner/database_cleaner#safeguards")
+        end
+      end
+
+      class ProductionEnv < Error
+        def initialize(env)
+          super("ENV['#{env}'] is set to production. Please refer to https://github.com/DatabaseCleaner/database_cleaner#safeguards")
+        end
+      end
+    end
+
+    class RemoteDatabaseUrl
+      def run
+        raise Error::RemoteDatabaseUrl if !skip? && given?
+      end
+
+      private
+
+        def given?
+          remote?(ENV['DATABASE_URL'])
+        end
+
+        def remote?(url)
+          url && !url.include?('localhost')
```

Comment:
What about `127.0.0.1`? Should that also be covered?

Reply:
@rwrede good call, thanks for the pointer! i've addressed that in https://github.com/svenfuchs/database_cleaner/commit/8b46dfcbbe7ee1bc0d83744a56b20c5dca480e74